rocAL/rocAL_pybind/examples/rocAL_api_python_unittest.py

```diff
@@ -52,154 +73,290 @@ def main():
     rocal_cpu = False if args.rocal_gpu else True
     device = "cpu" if rocal_cpu else "cuda"
     batch_size = args.batch_size
+    max_height = args.max_height
+    max_width = args.max_width
+    color_format = types.RGB if args.color_format else types.GRAY
+    tensor_layout = types.NHWC if args.color_format else types.NCHW
+    tensor_dtype = types.UINT8
     num_threads = args.num_threads
     random_seed = args.seed
-    local_rank =  args.local_rank
-    world_size =  args.world_size
-    display = True if args.display else False
+    local_rank = args.local_rank
+    world_size = args.world_size
+    interpolation_type = INTERPOLATION_TYPES[args.resize_interpolation_type]
+    scaling_mode = SCALING_MODES[args.resize_scaling_mode]
+    if (scaling_mode != types.SCALING_MODE_DEFAULT and interpolation_type !=
+            types.LINEAR_INTERPOLATION):
+        interpolation_type = types.LINEAR_INTERPOLATION
+    if augmentation_name in ["hue", "saturation", "color_twist"] and color_format == types.GRAY:
+        print("Not a valid option! Exiting!")
+        sys.exit(0)
+
     try:
-        path= "OUTPUT_IMAGES_PYTHON/NEW_API/FILE_READER/" + args.augmentation_name
+        path = "OUTPUT_IMAGES_PYTHON/NEW_API/FILE_READER/" + args.augmentation_name
         isExist = os.path.exists(path)
         if not isExist:
             os.makedirs(path)
     except OSError as error:
         print(error)
     # Create Pipeline instance
-    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=local_rank, seed=random_seed, rocal_cpu=rocal_cpu, tensor_layout=types.NHWC if args.NHWC else types.NCHW , tensor_dtype=types.FLOAT16 if args.fp16 else types.FLOAT)
+    pipe = Pipeline(batch_size=batch_size,
+                    num_threads=num_threads,
+                    device_id=local_rank,
+                    seed=random_seed,
+                    rocal_cpu=rocal_cpu,
+                    tensor_layout=tensor_layout,
+                    tensor_dtype=tensor_dtype,
+                    output_memory_type=types.CPU_MEMORY if rocal_cpu else types.GPU_MEMORY)
     # Set Params
     output_set = 0
     rocal_device = 'cpu' if rocal_cpu else 'gpu'
-    #hardcoding decoder_device to cpu until VCN can decode all JPEGs
+    # hardcoding decoder_device to cpu until VCN can decode all JPEGs
```

Comment:
@fiona-gladwin Is this still the case? With rocm5.6 all JPEGs should be enabled on Navi and MI cards

Reply:
Currently we are comparing the augmentation outputs with golden outputs taken with turbojpeg decoder so hardcoding to cpu for now. We will support both backends once the respective golden outputs are taken. I'll remove the "until VCN can decode all jpegs" part of the comment